packages/amplify_storage_s3/lib/method_channel_storage_s3.dart

```diff
@@ -168,23 +165,14 @@ class AmplifyStorageS3MethodChannel extends AmplifyStorageS3 {
 
   StorageException _prepareExceptionForMalformedResult(
       {@required String methodName, @required String fieldName}) {
-    const exceptionType =
-        StorageExceptionType.MALFORMED_PLATFORM_CHANNEL_RESULT;
-    final Map exceptionDetails = <String, String>{
-      'operation': methodName,
-      'malformed field': '$fieldName cannot be null'
-    };
     return StorageException(
-        code: exceptionType.code,
-        message: exceptionType.message,
-        details: exceptionDetails);
+      'Error formatting platform channel result',
+      recoverySuggestion: 'Operation $methodName failed: $fieldName cannot be null.');
   }
 
-  StorageException _convertToStorageException(
-      StorageExceptionType exceptionType, PlatformException exception) {
-    return StorageException(
-        code: exceptionType.code,
-        message: exception.message,
-        details: exception.details);
+  StorageException _convertToStorageException(PlatformException e) {
```

Comment:
Where is the `e.code` which is set to `StorageException` used?

What if it wasnt a `StorageException`?

Reply:
Right now everything coming from platform channel will become a StorageException. This method will expand to switch off the code once we add other exception types.